polynote-frontend/polynote/data/result.ts

```diff
@@ -286,8 +287,33 @@ export class ResultValue extends Result {
                     frag.appendChild(el);
                     return ["text/html", frag];
                 })
-            } else if (repr.knownSize && repr.knownSize > 0) {
-                // TODO: get a batch of data?
+            } else if (repr.knownSize && repr.knownSize > 0 && repr.knownSize < 1000) {
```

Comment:
The things about this is that `knownSize` is just the number of elements in the stream. Each element could be very large in which case we probably wouldn't want to do this...

The server already does an estimation of the total size of the encoded array – I think the solution would be to have the server send a `DataRepr` for small enough collections (instead of or in addition to the `StreamingDataRepr`)

Reply:
Makes sense - a `DataRepr` for something trivial like `Array[Int]` sounds good to me. 

For the general case of a `StreamingnDataRepr`, what do you think about fetching just the single batch then, if it gets to this case? Alternatively, we could do nothing for now and eventually pop out the table view or something similar that would allow users to page through the data.